pkg/cmd/stack/deploy.go

```diff
@@ -129,26 +129,68 @@ func deploy(ctx context.Context, s *model.Stack, c kubernetes.Interface, config
 			return
 		}
 
-		for _, name := range options.ServicesToDeploy {
-			if len(s.Services[name].Ports) > 0 {
-				svcK8s := translateService(name, s)
-				if err := services.Deploy(ctx, svcK8s, c); err != nil {
-					exit <- err
-					return
+		for _, serviceName := range options.ServicesToDeploy {
+			if len(s.Services[serviceName].Ports) == 0 {
+				continue
+			}
+
+			svcK8s := translateService(serviceName, s)
+			if err := services.Deploy(ctx, svcK8s, c); err != nil {
+				exit <- err
+				return
+			}
+			// get the public ports from the compose service - this will be deployed into ingresses
+			ingressPortsToDeploy := getSvcPublicPorts(serviceName, s)
+			for _, ingressPort := range ingressPortsToDeploy {
+				ingressName := serviceName
+				// If more than one port, ingressName will have <serviceName>-<PORT>, each port will have an ingress
+				if len(ingressPortsToDeploy) > 1 {
+					ingressName = fmt.Sprintf("%s-%d", serviceName, ingressPort.ContainerPort)
+				}
+				translateOptions := &ingresses.TranslateOptions{
+					Name:      s.Name,
+					Namespace: s.Namespace,
+				}
+
+				// merge endpoint labels and stack specific labels
+				endpointLabels := map[string]string{
+					model.StackNameLabel:         s.Name,
+					model.StackEndpointNameLabel: ingressName,
+				}
+				for k, v := range s.Services[serviceName].Labels {
+					endpointLabels[k] = v
 				}
-				ingressPorts := getSvcPublicPorts(name, s)
-				for _, ingressPort := range ingressPorts {
-					ingressName := name
-					if len(ingressPorts) > 1 {
-						ingressName = fmt.Sprintf("%s-%d", name, ingressPort.ContainerPort)
+
+				// parse an Endpoint model for this ingress
+				endpoint := model.Endpoint{
+					Labels:      endpointLabels,
+					Annotations: s.Services[serviceName].Annotations,
+					Rules: []model.EndpointRule{
+						{
+							Path:    "/",
+							Service: serviceName,
+							Port:    ingressPort.ContainerPort,
+						},
+					},
+				}
+
+				ingress := ingresses.TranslateEndpoint(ingressName, endpoint, translateOptions)
+
+				// check for labels collision in the case of a compose - before creation or update (deploy)
+				if old, _ := iClient.Get(ctx, ingress.GetName(), ingress.GetNamespace()); err != nil {
```

Comment:
I think we have to capture the error with `if old, err := ...`, if not we are checking the error from `iClient, err := ingresses.GetClient(ctx, c)`

Reply:
omg you are right! 😬 i will refactor a bit this part to encapsulate this check into a func